src/schedule/psk.rs

```diff
@@ -0,0 +1,65 @@
+#![allow(dead_code)]
+//! # Pre shared keys.
+
+/// enum {
+///   reserved(0),
+///   external(1),
+///   reinit(2),
+///   branch(3),
+///   (255)
+/// } PSKType;
+#[derive(Debug, PartialEq, Clone, Copy)]
+#[repr(u8)]
+pub(crate) enum PSKType {
+    Reserved = 0,
+    External = 1,
+    Reinit = 2,
+    Branch = 3,
+}
+
+struct ExternalPsk {
```

Comment:
Maybe add a TODO here, pointing to #141? `ExternalPSK` should probably be public and/or expose a public constructor. Also, some more documentation would be nice for the individual PSK types, but we can also leave that to #141.

Reply:
Yeah I'll add a TODO for the entire file. These are all really just dummy structs to actually be able to use `PreSharedKeys` as a type.